openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/core/TemplateOperationsImpl.java

```diff
@@ -185,8 +184,7 @@ public KubernetesList processLocally(Map<String, String> valuesMap) {
           } else if (Utils.isNotNullOrEmpty(parameter.getValue())) {
             parameterValue = parameter.getValue();
           } else if (EXPRESSION.equals(parameter.getGenerate())) {
-            Generex generex = new Generex(parameter.getFrom());
-            parameterValue = generex.random();
+            parameterValue = parameter.getFrom();
```

Comment:
This is not going to generate a valid template instance, at the same time:
- generating a random string can easily lead to errors
- the random string cannot be tracked

Alternatives are in:
1. substitute with an alternative library, but I'm not convinced about any of those:
   - https://github.com/curious-odd-man/RgxGen?tab=readme-ov-file
   - https://github.com/fmselab/mutrex
   - https://github.com/six2six/bfgex
   - https://github.com/agarciadom/xeger
 2. Roll our own implementation, either as an external micro library or within this project: the code is not going to be much, but OpenShift Templates are deprecated, so, we can even deprecate/remove this feature

Reply:
cc. @manusa @shawkins @metacosm

Reply:
> This is not going to generate a valid template instance

Why is this?

> - generating a random string can easily lead to errors
> - the random string cannot be tracked

The purpose of this method is really to process locally the templates as if they were being processed by the OpenShift cluster.

It really doesn't matter if the string can't be tracked or if the random string is invalid (maybe because the provided expression is wrong).
In the end, this would be the user's responsibility.

```yaml
parameters:
- description: Password used for Redis authentication
  from: '[A-Z0-9]{8}'
  generate: expression
  name: REDIS_PASSWORD
```
https://docs.openshift.com/container-platform/4.15/openshift_images/using-templates.html#templates-writing_using-templates


What we want is to be able to support this.

From the [docs](https://docs.openshift.com/container-platform/4.15/rest_api/template_apis/template-template-openshift-io-v1.html) we can read the constraints for this expression:

> generate specifies the generator to be used to generate random string from an input value specified by From field. The result string is stored into Value field. If empty, no generator is being used, leaving the result Value untouched. Optional.
>
> The only supported generator is "expression", which accepts a "from" value in the form of a simple regular expression containing the range expression "[a-zA-Z0-9]", and the length expression "a{length}".
> 
> Examples:
>
> | from | value |
> |------|-------|
> "test[0-9]{1}x" | "test7x"
> "[0-1]{8}" | "01001100"
> "0x[A-F0-9]{4}" | "0xB3AF"
> "[a-zA-Z0-9]{8}" | "hW4yQU5i"


It seems that a simple library or maybe even rolling our own utility method should suffice to handle this.